sympy/physics/units/util.py

```diff
@@ -154,6 +158,45 @@ def quantity_simplify(expr):
         ref = v[0]/v[0].scale_factor
         expr = expr.xreplace({vi: ref*vi.scale_factor for vi in v[1:]})
 
+    if across_dimensions:
+        # combine quantities of different dimensions into a single
+        # quantity that is equivalent to the original expression
+
+        unit_system = UnitSystem.get_unit_system(unit_system)
+        dimension_system: DimensionSystem = unit_system.get_dimension_system()
+        dim_expr = unit_system.get_dimensional_expr(expr)
+        dim_deps = dimension_system.get_dimensional_dependencies(dim_expr, mark_dimensionless=True)
+
+        target_dimension = None
+        for result_dim, result_deps in dimension_system.dimensional_dependencies.items():
+            if result_deps == dim_deps:
+                target_dimension = result_dim
+                break
+
+        from . import volt, ohm, farad, second, henry, meter, weber, siemens, tesla
+        from . import voltage, impedance, capacitance, inductance, magnetic_flux, conductance, magnetic_density, velocity
+
+        non_prefixed = unit_system.get_units_non_prefixed()
+        preferred_dimension_to_unit = { unit.dimension.name: unit for unit in non_prefixed if unit.dimension.name.is_symbol }
+
+        # HACK: add missing/wrong units to the map.
+        if unit_system.name == "SI":
+            preferred_dimension_to_unit.update({
```

Comment:
what about this:

```python
from sympy.physics.units import definitions

preferred_dimension_to_unit = {}

for name in dir(definitions):
    quantity = getattr(definitions, name)
    if not isinstance(quantity, Quantity):
        continue
    dim = unit_system.get_quantity_dimension(quantity)
    cc = convert_to(quantity, unit_system._base_units)
    if cc.as_coeff_Mul()[0] == 1:
        preferred_dimension_to_unit[dim.name] = quantity
```

Reply:
Unfortunately, this method appears to be more correct, but it absolutely kills performance. The unit test takes noticeably longer. Also, the velocity test case fails using this method. I think we can make something like this work, if we compute it beforehand. 

It's really annoying that there are all these prefixed units like `picojoule` and `teranewton`. IMO, they should not exist, we already have the prefixes defined using `Prefix`, that should be sufficient.

Reply:
Ultimately, I think that this is going to need some metadata changes, and _maybe_ breaking changes.

Reply:
What about using this method as a code generator? That dictionary needs to be computed only once in the end... Maybe when you install SymPy or when you use it the first time?

Reply:
I think that doing code gen is going to be harder than it sounds. An LRU cache seems like a better idea.

Reply:
> If an application using sympy were to allow for unescaped user input, it could be possible to end up saving and executing malicious code.

Well, the code generator should not be called by end users, but by the developers before releasing a new version of SymPy.

> Regardless, I've already implemented 2 new properties for Quantity: `is_prefixed` and `is_physics_constant`. With these, I'm pretty sure the call to `convert_to()` will not be necessary, and none of this will be required. I'll commit a new implementation shortly.

That's great to hear :)